saq/types.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+
+import typing as t
+
+from saq.compat import Literal, TypedDict
+
+if t.TYPE_CHECKING:
+    from saq.job import Job, Status
+
+BeforeEnqueueType = t.Callable[["Job"], t.Awaitable[t.Any]]
+Context = t.Dict[str, t.Any]
+CountKind = Literal["queued", "active", "incomplete"]
+DumpType = t.Callable[[t.Dict], str]
+DurationKind = Literal["process", "start", "total", "running"]
+Function = t.Callable
+ListenCallback = t.Callable[[str, "Status"], t.Any]
+LoadType = t.Callable[[t.Union[bytes, str]], t.Any]
+ReceivesContext = t.Callable[[Context], t.Awaitable[t.Any]]
+VersionTuple = t.Tuple[int, ...]
+
+
+class QueueInfo(TypedDict):
+    workers: dict
+    name: str
+    queued: int
+    active: int
+    scheduled: int
+    jobs: list[dict]
+
+
+class QueueStats(TypedDict):
+    complete: int
+    failed: int
+    retried: int
+    aborted: int
+    uptime: int
+
+
+class TimersDict(TypedDict, total=False):
+    schedule: int
+    stats: int
+    sweep: int
+    abort: int
```

Comment:
@tobymao Note in this module that I've used the typing versions of generics as it would be feasible that a user would import something from here to annotate their own code, and we can't be sure they'll use it with `from __future__ import annotations`, so needs to be compatible with 3.7 runtime.

This is also why `"Job"` and `"Status"` are forward refs here, as if this module is imported outside of a `if TYPE_CHECKING` block, their reference would otherwise fail.

Reply:
can you double check everything works fine in 3.7?

Reply:
yes I will verify

Reply:
`IsolatedAsyncioTestCase` not available in 3.7. I tried to do a compat with [aiounittest](https://aiounittest.readthedocs.io/en/latest/info.html) for 3.7 (`import aiounittest.AsyncTestCase as `IsolatedAsyncioTestCase`) but there were a stack of errors. 

Perhaps a simple integration test script can be added to CI for 3.7 only?

3.7 EOL is 27th June 2023.

Reply:
I've manually ported IsolatedAsyncioTestCase before. I can help with that.

@tobymao - is there some specific 3.7 use case you have in mind?

Reply:
i guess we can leave saq at 3.8+, in my mind everything i support is 3.7+ because of airflow, but i guess there's no reason to mix airflow and saq

Reply:
Then maybe we ignore 3.7?

Reply:
yes

Reply:
that would certainly simplify things

Reply:
3.7 removed and added 3.11

Reply:
Note we still need to use the typing module generics here unless we go back to:

```python
LoadType: t.TypeAlias = "Callable[[bytes | str], t.Any]"
```
Six to one, half a dozen to the other in my mind, so I'll leave as is unless you'd prefer otherwise.